client/wildcard/src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,181 @@
+
+            onOpenChange(event)
+        },
+        [isControlled, onOpenChange]
+    )
+
+    const context = useMemo(
+        () => ({
+            isOpen,
+            targetElement,
+            tailElement,
+            anchor,
+            setOpen,
+            setTargetElement,
+            setTailElement,
+        }),
+        [isOpen, targetElement, tailElement, anchor, setOpen]
+    )
+
+    return <PopoverContext.Provider value={context}>{children}</PopoverContext.Provider>
+}
+
+interface PopoverTriggerProps {}
+
+const PopoverTrigger = forwardRef((props, reference) => {
+    const { as: Component = 'button', onClick = noop, ...otherProps } = props
+    const { setTargetElement, setOpen, isOpen } = useContext(PopoverContext)
+
+    const callbackReference = useCallbackRef<HTMLButtonElement>(null, setTargetElement)
+    const mergedReference = useMergeRefs([reference, callbackReference])
+
+    const handleClick: React.MouseEventHandler<HTMLButtonElement> = event => {
+        setOpen({ isOpen: !isOpen, reason: PopoverOpenEventReason.TriggerClick })
+        onClick(event)
+    }
+
+    return <Component ref={mergedReference} onClick={handleClick} {...otherProps} />
+}) as ForwardReferenceComponent<'button', PopoverTriggerProps>
+
+interface PopoverContentProps extends Omit<FloatingPanelProps, 'target' | 'marker'> {
+    open?: boolean
+    focusLocked?: boolean
+}
+
+const PopoverContent = forwardRef((props, reference) => {
+    const {
+        open,
+        children,
+        focusLocked = true,
+        as: Component = 'div',
+        role = 'dialog',
+        'aria-modal': ariaModel = true,
+        ...otherProps
+    } = props
+
+    const { isOpen, targetElement, anchor, setOpen } = useContext(PopoverContext)
+
+    const localReference = useRef<HTMLDivElement>(null)
+    const mergeReference = useMergeRefs([localReference, reference])
+
+    // Catch any outside click of popover element
+    useOnClickOutside(mergeReference, event => {
+        if (targetElement?.contains(event.target as Node)) {
+            return
+        }
+
+        setOpen({ isOpen: false, reason: PopoverOpenEventReason.ClickOutside })
+    })
+
+    // Close popover on escape
+    useKeyboard({ detectKeys: ['Escape'] }, () => setOpen({ isOpen: false, reason: PopoverOpenEventReason.Esc }))
+
+    if (!isOpen && !open) {
+        return null
+    }
+
+    return (
+        <FloatingPanel
+            {...otherProps}
+            as={Component}
+            ref={mergeReference}
+            target={anchor?.current ?? targetElement}
+            role={role}
+            aria-modal={ariaModel}
+            className={classNames('dropdown-menu', otherProps.className)}
+        >
+            {focusLocked ? <FocusLock returnFocus={true}>{children}</FocusLock> : children}
+        </FloatingPanel>
+    )
+}) as ForwardReferenceComponent<'div', PopoverContentProps>
+
+const Root = Popover
+const Trigger = PopoverTrigger
+const Content = PopoverContent
```

Comment:
nit: because of this renaming jump to definition in VSCode opens the variable declaration instead of the `PopoverContent` definition which is probably the desired outcome. 

```suggestion
const Content = PopoverContent
```

Reply:
@valerybugakov I think I'm going to delete this variable re-exports anyway. See comments above https://github.com/sourcegraph/sourcegraph/pull/29791#discussion_r786760814